pkg/cli/admin/nodeimage/basenodeimagecommand.go

```diff
@@ -409,3 +441,37 @@ func (o *BaseNodeImageCommand) runNodeJoinerPod(ctx context.Context, tasks []fun
 	}
 	return nil
 }
+
+func (c *BaseNodeImageCommand) getIdmsFile(ctx context.Context) string {
+	imageDigestMirrorSets, idmsErr := c.ConfigClient.ConfigV1().ImageDigestMirrorSets().List(ctx, metav1.ListOptions{})
+	imageContentPolicies, icspErr := c.ConfigClient.ConfigV1().ImageContentPolicies().List(ctx, metav1.ListOptions{})
+	if idmsErr != nil && icspErr != nil {
+		return ""
+	}
+
+	// Build IDMS file from contents of the IDMS and ICSP mirror digests
+	var err error
+	var contents []byte
+	if idmsErr == nil && icspErr == nil {
+		contents, err = getIdmsContents(imageDigestMirrorSets, imageContentPolicies)
+	} else if idmsErr == nil && icspErr != nil {
+		contents, err = getIdmsContents(imageDigestMirrorSets, nil)
+	} else {
+		contents, err = getIdmsContents(nil, imageContentPolicies)
+	}
+	if err != nil {
+		c.log("failure parsing imageDigestMirrorSet %s", err.Error())
+		return ""
+	}
+
+	// create temp file which will be removed by caller
+	idmsFile, err := os.CreateTemp("", "idms-file")
+	c.log("Building IDMS file with contents %s", contents)
+	if _, err := idmsFile.Write(contents); err != nil {
+		idmsFile.Close()
+		os.Remove(idmsFile.Name())
+		return ""
```

Comment:
In this case I'd expect

```suggestion
		return "", err
```